ocp_utilities/monitoring.py

```diff
@@ -0,0 +1,209 @@
+import json
+import re
+from json import JSONDecodeError
+
+import requests
+from ocp_resources.route import Route
+from ocp_resources.secret import Secret
+from ocp_resources.service_account import ServiceAccount
+from ocp_resources.utils import TimeoutExpiredError, TimeoutSampler
+from simple_logger.logger import get_logger
+
+from ocp_utilities.infra import get_client
+
+
+TIMEOUT_2MIN = 2 * 60
+TIMEOUT_10MIN = 10 * 60
+
+LOGGER = get_logger(name=__name__)
+
+
+class Prometheus(object):
+    """
+    For accessing Prometheus cluster metrics
+
+    Prometheus HTTP API doc:
+    https://prometheus.io/docs/prometheus/latest/querying/api/
+
+    Argument for query method should be the entire string following the server address
+        e.g.
+        prometheus = Prometheus()
+        up = prometheus.query("/api/v1/query?query=up")
+    """
+
+    def __init__(
+        self, namespace=None, resource_name=None, client=None, verify_ssl=True
+    ):
+        self.namespace = namespace or "openshift-monitoring"
+        self.resource_name = resource_name or "prometheus-k8s"
+        self.client = client or get_client()
+        self.api_v1 = "/api/v1"
+        self.verify_ssl = verify_ssl
+
+        route = Route(
```

Comment:
Please extract `reoute` to method of the class and use it.

Reply:
done

Reply:
created the method to extract route, please take look a the latest changes